### ISSUE TITLE
[HOLD for payment 2023-09-20] [$1000] Inconsistent validation of name in workspace settings

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#23376]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Login to new dot
2. Open settings
3. Click on workspaces and select any workspace. Create one if there's none present.
4. Click on settings.
5. Try to change the name of workspace to `<lol>`. Observe the error after clicking on save.
6. Now try to change the name to `< lol>`.
7. Click on save.
8. Now again try to change the name to `<>`.
## Expected Result: 
We shouldn't be able to save to any of the following and error should be shown immediately after clicking save without going back.
## Actual Result: 
For `<lol>`, it is the ideal behaviour
For `<>`, it actually gets saved without any problem
For `< lol>`, it didn't show error immediately, and goes back to workspace page.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.44-0
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856

https://github.com/Expensify/App/assets/43996225/b17e5722-454a-4eef-b1b8-960c4263cf66

https://github.com/Expensify/App/assets/43996225/37763e02-bef8-4133-9b3f-5018042d0469

**Notes/Photos/Videos:** Any additional supporting documentation

**Expensify/Expensify Issue URL:**
**Issue reported by:** @shubham1206agra
**Slack conversation:**  https://expensify.slack.com/archives/C049HHMV9SM/p1689605016201359

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0174a929d45c708370</li>
        <li>Upwork Job ID: 1683849578868912128</li>
        <li>Last Price Increase: 2023-08-15</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Reporter | 26189493</li>
</ul></li>
    </ul>
</details>

**Comment by zanyrenney** [comment]
Didn't get to testing this today. Will prioritise tomorrow!

**Comment by shubham1206agra** [comment]
Hey @zanyrenney 
Can you check in the backend for the regex that WAF is using and paste it here?

**Comment by zanyrenney** [comment]
I do not understand what you are asking. @shubham1206agra

**Comment by vadymbokatov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Validation for HTML tag name is not working correctly in workspace settings page.
### What is the root cause of that problem?
Regex of HTML tag name is not matching. If there is space in <> or there is nothing in <>.
`VALIDATE_FOR_HTML_TAG_REGEX: /<([^>\s]+)(?:[^>]*?)>/g`
### What changes do you think we should make in order to solve the problem?

We need to update this regex to contain all the cases.
like this:
`VALIDATE_FOR_HTML_TAG_REGEX: /<([^>\s]*)(?:[^>]*?)>/g`
### What alternative solutions did you explore? (Optional)
### Result

https://github.com/Expensify/App/assets/138146362/2a0a5742-402c-49ce-9d38-04253e5df944

**Comment by zanyrenney** [comment]
@shubham1206agra please could you, it's why you are looking for it?
the WAF has hundreds of regexes that get run but if you think something is getting filtered out we'll have logs that show that. Thanks!

**Comment by robertKozik** [comment]
@zanyrenney I think @shubham1206agra point of getting this regex from backend is valid one. Both these regexes should be the same to make sure this kind of issues. 
I think the point of this issue would be to unify these two regexes so the validation would be the same on FE and BE

**Comment by shubham1206agra** [comment]
> @shubham1206agra please could you, it's why you are looking for it? the WAF has hundreds of regexes that get run but if you think something is getting filtered out we'll have logs that show that. Thanks!

@zanyrenney For the case of `< lol>`, the error comes from backend. So I think I need to know if the regex changed in backend at some point in time.

And we also need to see `< >` is escaping from backend.

And this regex has problem in multiple area.

**Comment by vadymbokatov** [comment]
@shubham1206agra I think, we don't need to update the regex in the backend. Just update regex on the front end

**Comment by shubham1206agra** [comment]
> @vadymbokatov After setting the name to "< abcdefg>", UpdateWorkspaceGeneralSettings is called and returns:
> 
> ![image](https://user-images.githubusercontent.com/137707942/256297468-31e8db1e-e902-4616-92be-b4a5bc55028e.png)

Is this response of API request?

**Comment by shubham1206agra** [comment]
Can you expand the response and send it here?

**Comment by shubham1206agra** [comment]
> Sure, here it is:
> 
> ```json
> {
>     "code": 666,
>     "jsonCode": 666,
>     "type": "Expensify\\Libs\\Error\\ExpError",
>     "UUID": "7442A242-6C3F-4C7C-90A5-D3A5352527FA",
>     "message": "Invalid workspace name",
>     "title": "",
>     "data": {
>         "onyxData": [
>             {
>                 "onyxMethod": "merge",
>                 "key": "workspaceSettingsForm",
>                 "value": {
>                     "errorFields": {
>                         "name": {
>                             "1690388067561870": "Invalid workspace name"
>                         }
>                     }
>                 }
>             }
>         ]
>     },
>     "htmlMessage": "",
>     "onyxData": [
>         {
>             "onyxMethod": "merge",
>             "key": "workspaceSettingsForm",
>             "value": {
>                 "errorFields": {
>                     "name": {
>                         "1690388067561870": "Invalid workspace name"
>                     }
>                 }
>             }
>         }
>     ],
>     "requestID": "7ecdef8dac070e70-AMS"
> }
> ```

Yeah
Getting same response

**Comment by robertKozik** [comment]
As far as I see, this issue is related to the backend - the regex or the processing of the name string from workspace settings. To fully fix this issue, we will need insights into the regex and what is done with the name string beforehand. If we cannot get this information, I believe this issue should become internal as an internal engineer can look into both the newdot code and backend code to unify the validation here.
CC. @zanyrenney

**Comment by zanyrenney** [comment]
Thank you for your thoughts @robertKozik ! I really appreciate. I will instead apply the internal label so that we get on Expensify engineer who can look at what is happening here.

**Comment by techievivek** [comment]
Thanks for all the discussions so far, it was very helpful.

It looks like we have the following cases:

```
<abc> - Getting blocked in the front end itself. (I can confirm this would be blocked in the backend as well, we will strip XSS vectors so this would be reduced to an empty string which we don't allow for workspace name).
< abc> - Frontend allows it, but the backend blocks it. (We are throwing an error in the backend because our XSS WAF rule converts the value to an empty string, and since we don't allow the empty string as a value for the workspace name, this no longer works.
<> -> Both frontend and backend allow it. (Backend allows it because we have whitelisted this input in our XSS WAF rule as it is a safe input if XSS is to be considered).
```

**Comment by techievivek** [comment]
Bought a discussion internally to learn more about our naming convention for the workspace. Once we all agree on something, I can share them here and we can make the changes accordingly.

**Comment by techievivek** [comment]
Ok, so after the discussion we have decided that for
1. `<>` we want to keep behaviour as it is, that is, allow workspaces to name to be `<>`. Similar to `<>` we have some other special values that can be valid workspace names. `'<>', '< >', '<->', '<-->', '<br>', '<br/>'`.(Maybe we can create a const to hold these values).
2. Fix the regex rule to catch inputs like `< lol>` in frontend.

We have rules that filter out any input values that are XSS sensitive in the backend, except for certain special values listed in point 1. To keep things consistent, let's update the regex on the frontend to do the same. Let me know if you have any questions, thanks.

**Comment by techievivek** [comment]
Yes, waiting for proposals from contributors.

**Comment by shubham1206agra** [comment]
> Ok, so after the discussion we have decided that for
> 
>     1. `<>` we want to keep behaviour as it is, that is, allow workspaces to name to be `<>`. Similar to `<>` we have some other special values that can be valid workspace names. `'<>', '< >', '<->', '<-->', '<br>', '<br/>'`.(Maybe we can create a const to hold these values).
> 
>     2. Fix the regex rule to catch inputs like `< lol>` in frontend.
> 
> 
> We have rules that filter out any input values that are XSS sensitive in the backend, except for certain special values listed in point 1. To keep things consistent, let's update the regex on the frontend to do the same. Let me know if you have any questions, thanks.

Can you post the regex here?
Thanks

**Comment by techievivek** [comment]
Just confirming with the team if it is ok to share the regex in public or not: https://expensify.slack.com/archives/C03TQ48KC/p1691664773190499, thanks. But in short, we want to update the regex so it take into account the leading space before the tag name.

**Comment by sourcecodedeveloper** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Workspace names are allowing leading space before the tag name.

### What is the root cause of that problem?
The regex rule below does not account for leading space before tag name

```inputValue.search(CONST.VALIDATE_FOR_HTML_TAG_REGEX) ```


### What changes do you think we should make in order to solve the problem?
For the changes as mentioned by @techievivek from point 1, the special cases which are considered to be valid workspace names are covered in a const array ignoreRegList. 

```
const ignoreRegxList = [/<>/, /< >/, /<->/, /<-->/, /<br>/, /<br\/>/];
const isMatch = ignoreRegxList.some(r => r.test(inputValue));
if(isMatch)
{
    return;
}
```

From @techievivek point 2,
I have made another regex to cover the case for leading spaces before tag names. 

```
const leadingSpaceRegex= inputValue.search(/<(\s+[A-Za-z0-9]+)>/g);
const m = inputValue.search(CONST.VALIDATE_FOR_HTML_TAG_REGEX);

// Return early if there is no value OR the value is not a string OR there are no HTML characters
if (!inputValue || !_.isString(inputValue) || (m && leadingSpaceRegex)  === -1) {
    return;
}

```


https://github.com/Expensify/App/assets/141340971/804f4e11-6d4a-44af-ae36-a10d4b62ce24

**Comment by sourcecodedeveloper** [comment]
> Yes, waiting for proposals from contributors.

@techievivek please review [proposal](https://github.com/Expensify/App/issues/23376#issuecomment-1676451184)

**Comment by techievivek** [comment]
@sourcecodedeveloper Looks good, I guess we are not updating the original regex because it is used at other places? CC @robertKozik, can you please review the proposals? Thanks.

**Comment by sourcecodedeveloper** [comment]
> @sourcecodedeveloper Looks good, I guess we are not updating the original regex because it is used at other places? CC @robertKozik, can you please review the proposals? Thanks.

@techievivek the original regex is functioning fine short of not capturing leading spaces before tag names. It is better to leave the original regex as is and simply add another regex to handle leading spaces and the special values are covered in the const array.

**Comment by shubham1206agra** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.

Inconsistent validation of name in workspace settings

### What is the root cause of that problem?

Root cause of the problem is this validation in `Form`

https://github.com/Expensify/App/blob/9004c3968fcde7f46a473db871b4dd6ab2f3effc/src/components/Form.js#L126

This regex here

https://github.com/Expensify/App/blob/9004c3968fcde7f46a473db871b4dd6ab2f3effc/src/CONST.js#L762

Matches only if there is at least one non-space character before >.

### What changes do you think we should make in order to solve the problem?

We can change this regex to

```js
/<((?!(br\/?>)|(\s?>)|(--?>))+)(?:[^>]*?)>/g
```

so it matches according to @techievivek comment https://github.com/Expensify/App/issues/23376#issuecomment-1667497623

Test screenshot

<img width="697" alt="image" src="https://github.com/Expensify/App/assets/58412969/8baf9d9c-b9c4-4a08-9ac8-5ecc42577e5a">


### What alternative solutions did you explore? (Optional)

**Comment by techievivek** [comment]
I bumped @robertKozik in [Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1692162636087309) to prioritize reviewing the proposals here.

**Comment by shubham1206agra** [comment]
> I suggest to put this on hold for #21133. That issue is waiting for decision here - [#21133 (comment)](https://github.com/Expensify/App/issues/21133#issuecomment-1674550063)

This issue is a little different from what @aimane-chnaif referenced.

Another issue with the same RCA as this issue
https://expensify.slack.com/archives/C049HHMV9SM/p1690611987275289
https://expensify.slack.com/archives/C049HHMV9SM/p1690306702107469

**Comment by techievivek** [comment]
I discussed this with Aimane [here](https://expensify.slack.com/archives/C02NK2DQWUX/p1692162636087309) and both are different issue so let's keep this moving.

**Comment by robertKozik** [comment]
Hi all! I've checked both proposal which came after the clarification from @techievivek - so the one from @shubham1206agra and from @sourcecodedeveloper.

> `<>`, `< >`, `<->`, `<-->`, `<br>`, `<br/>`

@techievivek, is it possible that this list might change in the future? I'm asking because if there are plans to expand or reduce this list later on, it would be advisable to store it as a constant array as suggested and split the implementation as suggested in @sourcecodedeveloper proposal. On the other hand, if the existing regex VALIDATE_FOR_HTML_TAG_REGEX is currently utilized only in one place, my preference would be to keep all aspects encompassed within a single regex for simplicity.

**Comment by shubham1206agra** [comment]
> Hi all! I've checked both proposal which came after the clarification from @techievivek - so the one from @shubham1206agra and from @sourcecodedeveloper.
> 
> > `<>`, `< >`, `<->`, `<-->`, `<br>`, `<br/>`
> 
> @techievivek, is it possible that this list might change in the future? I'm asking because if there are plans to expand or reduce this list later on, it would be advisable to store it as a constant array as suggested and split the implementation as suggested in @sourcecodedeveloper proposal. On the other hand, if the existing regex VALIDATE_FOR_HTML_TAG_REGEX is currently utilized only in one place, my preference would be to keep all aspects encompassed within a single regex for simplicity.

If the list changes in the BE, we wouldn't know unless someone raises an issue about this in future or someone from the internal team changes it from frontend also.

**Comment by sourcecodedeveloper** [comment]
> Hi all! I've checked both proposal which came after the clarification from @techievivek - so the one from @shubham1206agra and from @sourcecodedeveloper.
> 
> > `<>`, `< >`, `<->`, `<-->`, `<br>`, `<br/>`
> 
> @techievivek, is it possible that this list might change in the future? I'm asking because if there are plans to expand or reduce this list later on, it would be advisable to store it as a constant array as suggested and split the implementation as suggested in @sourcecodedeveloper proposal. On the other hand, if the existing regex VALIDATE_FOR_HTML_TAG_REGEX is currently utilized only in one place, my preference would be to keep all aspects encompassed within a single regex for simplicity.

The original regex VALIDATE_FOR_HTML_TAG_REGEX is being maintained on the backend. According to the comment below the regex on the front end is to match with that on the backend. The comment explains the reason why the regex on the front end has to match with the server i.e to be consistent. 

```
It's copied here so that the same regex pattern can be used in form validations to be consistent with the server.
```
```
    // The server has a WAF (Web Application Firewall) which will strip out HTML/XML tags using this regex pattern.
    // It's copied here so that the same regex pattern can be used in form validations to be consistent with the server.
    VALIDATE_FOR_HTML_TAG_REGEX: /<([^>\s]+)(?:[^>]*?)>/g,
```

Changing the original regex on the front end would create inconsistency with the regex on the backend.

**Comment by shubham1206agra** [comment]
> > Hi all! I've checked both proposal which came after the clarification from @techievivek - so the one from @shubham1206agra and from @sourcecodedeveloper.
> > 
> > > `<>`, `< >`, `<->`, `<-->`, `<br>`, `<br/>`
> > 
> > @techievivek, is it possible that this list might change in the future? I'm asking because if there are plans to expand or reduce this list later on, it would be advisable to store it as a constant array as suggested and split the implementation as suggested in @sourcecodedeveloper proposal. On the other hand, if the existing regex VALIDATE_FOR_HTML_TAG_REGEX is currently utilized only in one place, my preference would be to keep all aspects encompassed within a single regex for simplicity.
> 
> The original regex VALIDATE_FOR_HTML_TAG_REGEX is being maintained on the backend. According to the comment below the regex on the front end is to match with that on the backend. The comment explains the reason why the regex on the front end has to match with the server i.e to be consistent. 
> 
> ```
> It's copied here so that the same regex pattern can be used in form validations to be consistent with the server.
> ```
> ```
>     // The server has a WAF (Web Application Firewall) which will strip out HTML/XML tags using this regex pattern.
>     // It's copied here so that the same regex pattern can be used in form validations to be consistent with the server.
>     VALIDATE_FOR_HTML_TAG_REGEX: /<([^>\s]+)(?:[^>]*?)>/g,
> ```
> 
> Changing the original regex on the front end would create inconsistency with the regex on the backend.
> 

Actually regex got changed
That's why it's not working in the first place
And as per @techievivek guidelines, this is the most fitting regex I can think of without knowing the exact regex in the backend

**Comment by techievivek** [comment]
@robertKozik 

>  is it possible that this list might change in the future?

This hasn't changed in years so most likely we will not need to update it in future. And in the backend, we have a similar structure where we have a constant whitelisting these specials values and a single regex that validates the rest of the inputs. 

I would suggest we do the same. Have a const to whitelist above special values and a single regex that validates the HTML inputs.

**Comment by shubham1206agra** [comment]
## Proposal

[Updated](https://github.com/Expensify/App/issues/23376#issuecomment-1677521687)

**Comment by robertKozik** [comment]
thanks for clarifying @techievivek, If thats how its handled on BE we should imitate it on FE also. I've checked the regex and solution from @sourcecodedeveloper, it tests well and checks out all scenarios pointed out by @techievivek. As it was also the first after updating the conditions of the issue I'll go with that solution

CC. @techievivek

**Comment by shubham1206agra** [comment]
> thanks for clarifying @techievivek, If thats how its handled on BE we should imitate it on FE also. I've checked the regex and solution from @sourcecodedeveloper, it tests well and checks out all scenarios pointed out by @techievivek. As it was also the first after updating the conditions of the issue I'll go with that solution
> 
> CC. @techievivek

This solution has problems which will cause regressions.

**Comment by shubham1206agra** [comment]
Test for this string `'<>< ><--><-><br><br/><lol>< lol>'`

**Comment by techievivek** [comment]
@robertKozik You will have to add a comment similar to [this](https://github.com/Expensify/App/issues/24569#issuecomment-1680060413) to trigger the automation. (ribbon and eyes combo).

**Comment by robertKozik** [comment]
Ok, going to check it with this one

**Comment by robertKozik** [comment]
Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me:
<img width="379" alt="image" src="https://github.com/Expensify/App/assets/61146594/60bcd7c8-6141-4240-ab35-541408d93754">

I'll check the next proposal then so we hopefully won't be having any regression problem

CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE

**Comment by robertKozik** [comment]
> @robertKozik You will have to add a comment similar to [this](https://github.com/Expensify/App/issues/24569#issuecomment-1680060413) to trigger the automation. (ribbon and eyes combo).

Sure! Will remember about it!

**Comment by sourcecodedeveloper** [comment]
> Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me: <img alt="image" width="379" src="https://user-images.githubusercontent.com/61146594/261284540-60bcd7c8-6141-4240-ab35-541408d93754.png">
> 
> I'll check the next proposal then so we hopefully won't be having any regression problem
> 
> CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE

@robertKozik valid names according to the const array should be <>< ><--><-><br><br/>  not including <lol>< lol>
i can check and if it is letting through I can update proposal and update the regex.

Also to be fair, I was the first to propose use of const array compared to the proposal by @shubham1206agra whose proposal was to update the original regex. If there is simple lack in fixing regex I am more than happy to update proposal.

**Comment by shubham1206agra** [comment]
> Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me: <img alt="image" width="379" src="https://user-images.githubusercontent.com/61146594/261284540-60bcd7c8-6141-4240-ab35-541408d93754.png">
> 
> I'll check the next proposal then so we hopefully won't be having any regression problem
> 
> CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE

Then I think it's a big problem on BE side
Cause I can inject `script` tags as long as I have one tag from the exclusion list

**Comment by sourcecodedeveloper** [comment]
> > Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me: <img alt="image" width="379" src="https://user-images.githubusercontent.com/61146594/261284540-60bcd7c8-6141-4240-ab35-541408d93754.png">
> > I'll check the next proposal then so we hopefully won't be having any regression problem
> > CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE
> 
> @robertKozik valid names according to the const array should be `<>< ><--><-><br><br/>` not including `<lol>< lol>` i can check and if it is letting through I can update proposal and update the regex.
> 
> Also to be fair, I was the first to propose use of const array compared to the proposal by @shubham1206agra whose proposal was to update the original regex. If there is simple lack in fixing regex I am more than happy to update proposal.

@robertKozik I just checked and I am able to update the regex in the frontend to not allow '<lol> < lol>'

**Comment by shubham1206agra** [comment]
> > @robertKozik You will have to add a comment similar to [this](https://github.com/Expensify/App/issues/24569#issuecomment-1680060413) to trigger the automation. (ribbon and eyes combo).
> 
> Sure! Will remember about it!

I am getting filtered to `<>< ><--><-><br><br/>` @robertKozik if I let it pass through

**Comment by shubham1206agra** [comment]
> Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me: <img alt="image" width="379" src="https://user-images.githubusercontent.com/61146594/261284540-60bcd7c8-6141-4240-ab35-541408d93754.png">
> 
> I'll check the next proposal then so we hopefully won't be having any regression problem
> 
> CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE

I am getting filtered to `<>< ><--><-><br><br/>` @robertKozik  if I let it pass through

**Comment by techievivek** [comment]
@sourcecodedeveloper already mentioned that in the comment [here](https://github.com/Expensify/App/issues/23376#issuecomment-1682066142), what it is doing is that if the input contains values that are XSS prone and are not whitelisted then they get stripped.

**Comment by shubham1206agra** [comment]
> @sourcecodedeveloper already mentioned that in the comment [here](https://github.com/Expensify/App/issues/23376#issuecomment-1682066142), what it is doing is that if the input contains values that are XSS prone and are not whitelisted then they get stripped.

Yeah
I have tried to put `script` tag. Now no API call was made somehow for this update. But if I do anything else, API call will occur

**Comment by sourcecodedeveloper** [comment]
> Hmm yeah you're right @shubham1206agra, the already existing solution is not letting through the `<>< ><--><-><br><br/><lol>< lol>` but the one from proposal is letting it through. But on the other hand the backend is saving such a name without throwing an error for me: <img alt="image" width="379" src="https://user-images.githubusercontent.com/61146594/261284540-60bcd7c8-6141-4240-ab35-541408d93754.png">
> 
> I'll check the next proposal then so we hopefully won't be having any regression problem
> 
> CC. @techievivek as I'm not sure `<>< ><--><-><br><br/><lol>< lol>` should be valid name in BE

@robertKozik I have updated code it now runs the original regex first which catches the `<lol> < lol>` consistent with the regex on server side. attaching screenshot below 

![image](https://github.com/Expensify/App/assets/141340971/e8512d32-3939-41d5-aea4-c5734cb26bc5)

**Comment by kosmydel** [comment]
Hey, I think we can use a similar approach to the chat messages, where we can send `<htmltags>` correctly.

More details in [my proposal](https://github.com/Expensify/App/issues/21133#issuecomment-1666457954) in another issue, which apparently addresses this issue as well.

In that way, users would be able to use all `<htmltags>` in workspace names.

In short:
- we can escape the workspace name on the FE side, before validation, to bypass WAF,
- sent escaped version of workspace name in HTML responses (image -> 1),
- send an unescaped version of the workspace name in other responses (image -> 2).

![image](https://github.com/Expensify/App/assets/104823336/2dd71e9b-f1b2-41ba-81e4-810fecf05bdb)

**Comment by sourcecodedeveloper** [comment]
> I doubt the goal is to bypass or go around WAF by sending the escaped version!

**Comment by robertKozik** [comment]
Thanks for updating the proposal @sourcecodedeveloper, your proposal was first to store the whitelisted tags as array to imitate the backend flow there, it should be beneficial in fixing any of further inconsistents. That's also why I think we should go with your proposal rather than one powerful regex proposed by @shubham1206agra, in this case unifying the logic between two ends should be prioritised.
I'm opting for the @sourcecodedeveloper [proposal](https://github.com/Expensify/App/issues/23376#issuecomment-1676451184) as it was first to introduce similar flow to a BE validation flow, and needed only small adjustments after to minimise the regression possibility.

🎀 👀 🎀 C+ reviewed

**Comment by shubham1206agra** [comment]
> Thanks for updating the proposal @sourcecodedeveloper, your proposal was first to store the whitelisted tags as array to imitate the backend flow there, it should be beneficial in fixing any of further inconsistents. That's also why I think we should go with your proposal rather than one powerful regex proposed by @shubham1206agra, in this case unifying the logic between two ends should be prioritised. I'm opting for the @sourcecodedeveloper [proposal](https://github.com/Expensify/App/issues/23376#issuecomment-1676451184) as it was first to introduce similar flow to a BE validation flow, and needed only small adjustments after to minimise the regression possibility.
> 
> 🎀 👀 🎀 C+ reviewed

I do have another regression issue with the current @sourcecodedeveloper proposal. Since @sourcecodedeveloper hasn't updated the code in the proposal, I can't fully confirm it.

**Comment by shubham1206agra** [comment]
@robertKozik You have seen my alternate proposal, right?

**Comment by sourcecodedeveloper** [comment]
@robertKozik @techievivek  Thanks for accepting the proposal! The PR has been created [#25530](https://github.com/Expensify/App/pull/25530)
I am still going through the long list of steps and it should ready for review shortly.

**Comment by sourcecodedeveloper** [comment]
@robertKozik @techievivek PR is ready and waiting to be reviewed!

**Comment by robertKozik** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [X]  [@robertKozik] The PR that introduced the bug has been identified. Link to the PR: **It was inconsistent behaviour between validation on FE and BE, no PR to blame on.**
> * [X]  [@robertKozik] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **N/A**
> * [X]  [@robertKozik] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**
> * [X]  [@robertKozik] Determine if we should create a regression test for this bug. **I believe there's no need to create a regression test for this. The validation will remain consistent with the backend unless it undergoes a change. It's difficult to reliably catch this with regression tests.**
> * [X]  [@robertKozik] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. **N/A**

**Comment by zanyrenney** [comment]
payout summary 

@robertKozik does not require payment (Contractor) **not paid here**
@sourcecodedeveloper requires payment
@shubham1206agra requires payment [offer](https://www.upwork.com/nx/wm/offer/26189493) (Reporter)

**Comment by zanyrenney** [comment]
@sourcecodedeveloper assuming your upwork profile is: AgileDeveloper K.
@shubham1206agra - offer sent!

**Comment by zanyrenney** [comment]
bump @sourcecodedeveloper please confirm the above.

**Comment by zanyrenney** [comment]
@robertKozik does not require payment (Contractor) not paid here
@sourcecodedeveloper requires payment - PAID.
@shubham1206agra requires payment [offer](https://www.upwork.com/nx/wm/offer/26189493) (Reporter) - PAID.
